c10/core/DispatchKeySet.cpp

```diff
@@ -12,10 +12,46 @@ constexpr DispatchKeySet autograd_dispatch_keyset = DispatchKeySet({
   DispatchKey::AutogradOther,
 });
 
+constexpr DispatchKeySet math_dispatch_keyset = DispatchKeySet({
```

Comment:
We shouldn't need to duplicate these big slabs - ideally there'd be a single source of truth for each alias keyset. I'd say let's build the `DispatchKeySet`s from the `std::array`s defined further down, and have the arrays be the source of truth. (If there's some reason that doesn't work straightforwardly, maybe that's another reason to go to macros...)

Reply:
actually the iterator for DispatchKeySet landed so I'll rebase and remove the std::array one ;)

Reply:
😍